handler.go

```diff
@@ -130,30 +133,31 @@ func (h *handler) fillByCachedUserContexts(contexts []*context.UserContext, reqU
 	return userContexts, nil
 }
 
-type optionsHandler struct {
-	methods []string
-}
+const ErrorMessageForMethodNotAllowed = "request resource does not support http method '%s'"
 
-func newOptionsHandler() *optionsHandler {
-	return &optionsHandler{
-		methods: make([]string, 0),
-	}
+func notAllowHandler(method string) router.Handler {
+	return newHandler(reflect.ValueOf(func() *response.Response {
+		return response.New(fmt.Sprintf(ErrorMessageForMethodNotAllowed, method)).Status(http.StatusMethodNotAllowed)
+	}))
 }
 
-func (o *optionsHandler) addMethod(method string) *optionsHandler {
-	o.methods = append(o.methods, method)
-	return o
-}
+type optionsHandler struct{}
 
-func (o *optionsHandler) build() *handler {
-	allowMethods := "OPTIONS"
-	for _, m := range o.methods {
-		allowMethods += ", " + m
-	}
+func (o *optionsHandler) Build(allowMethods string) router.Handler {
 	return newHandler(reflect.ValueOf(func() *response.Response {
 		return response.New("").
 			Headers(response.Headers{
 				"Allow": allowMethods,
 			})
 	}))
 }
+
+func (h *handler) getRoute() string {
+	return h.route
+}
+
+// WildcardIndex implements router.Handler
+func (h *handler) WildcardIndex(i int) error {
```

Comment:
If this would never return error, I think we should remove the return value.

Reply:
Please take a look at the `router.Handler` interface, I left a comment about why I allow it returns an error, furthermore I'm going to remove this from the interface for the following reasons:
- it depends on the implementation of `router`, `router` split path by `/` for now, but future?
- it's hard to understand the purpose, easy to misuse it

or you can say those issues are the same one: current design make `router` and `handler` depends on another's implementation details.

For now, I won't change this, I think we need another issue to think about the problem.